virttest/env_process.py

```diff
@@ -1045,6 +1045,11 @@ def preprocess(test, params, env):
             kernel_extra_params_add += " intel_iommu=on"
         else:
             kernel_extra_params_remove += " intel_iommu=on"
+        guest_iommu_pt = params.get("guest_iommu_pt")
```

Comment:
I don't have very strong feeling about this.  guest `iommu` support not only one option, but `iommu=pt` be used more often in our test.  Prefer this a little more:  define parameter `guest_iommu_option = $option`, if some option were added, add it to `kernel_extra_params_add` with `iommu = $option`.  
refer the info from kernel doc:
```
    iommu=      [x86]                                                               
        off                                                                         
        force                                                                       
        noforce                                                                     
        biomerge                                                                    
        panic                                                                       
        nopanic                                                                     
        merge                                                                       
        nomerge                                                                     
        soft                                                                        
        pt      [x86, IA-64]                                                        
        nobypass    [PPC/POWERNV]                                                   
            Disable IOMMU bypass, using IOMMU for PCI devices. 
```

@jingzhao84 @luckyh call for more opinions.

Reply:
Not all platforms have the option `pt`, so I think @yfu 's suggestion is quite good.

Reply:
@yanan-fu @luckyh Yes, that's better. Just I will need remove the other `iommu=\w+` options when add one(e.g. replace `iommu=off` with `iommu=pt`), this way I might need rewrite `check_kernel_cmdline` for fuzzy matching(not confident, seems also need update all caller, since need a list instead of a string with space for `kernel_extra_params_remove`, since it will mix string and regular expression in the list), or I might get the accurate string from `"cat /proc/cmdline"` first and add it to `kernel_extra_params_remove` directly(easier approach), which one do you prefer? Or any other suggestion?

Reply:
@vivianQizhu  Good question.  According to this thinking,  when we use `guest_iommu_pt == "yes"`, we need to check other iommu option first too i think.
1. `guest_iommu_pt = yes`,  automation doesn't provide method to add other iommu option, so check other options are not that must ?
2. `guest_iommu_option = pt`, automation provide method to add other option, so check other exclusive options is must ?

Reply:
@yanan-fu If we don't consider other options I think current solution would be enough, but if we consider all option, we need consider more. I was thinking append an parameter like `update_args` to `update_boot_option`, input `iommu: pt` and in the function replace the option with the desired one if it exists, otherwise add it, but this way I will need update `update_boot_option_ubuntu` too...So I now prefer adding an API in `utils_test` like `get_kernel_option(option_name)` returns option value.

Reply:
@yanan-fu Just found that `kernel_extra_params_add` will finally used as `grubby --update-kernel=`grubby --default-kernel` --args="$kernel_extra_params_add"`, per grubby manual page: `If an argument already exists the new value replaces the old values`, I have tried and it does replace old one. So do no need extra handling in our code for the update issue. Therefore I updated to only append defined option to `kernel_extra_params_add`. Please help review, thanks.

Reply:
IIRC, `grubby` would handle the replacement, that means for the case user wants `iommu=X` it will automatically replace the former value with the new one `X`, so that may not be a problem.

The question is what the param `guest_iommu_*` at here stand for, if it exactly represent the kernel command line option `iommu`, then it should be a enum, not a bool; if it is an abstraction of the user's requirement to set the iommu mode to `pt`, then it can be a bool, and we should provide the usage to users.

Thanks.

Reply:
@luckyh You are right, and I have updated the code, please help review, thanks.